server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -161,18 +181,116 @@ public TextFieldMapper build(BuilderContext context) {
                     builder.fielddataFrequencyFilter(minFrequency, maxFrequency, minSegmentSize);
                     DocumentMapperParser.checkNoRemainingFields(propName, frequencyFilter, parserContext.indexVersionCreated());
                     iterator.remove();
+                } else if (propName.equals("index_prefix")) {
+                    Map<?, ?> indexPrefix = (Map<?, ?>) propNode;
+                    int minChars = XContentMapValues.nodeIntegerValue(indexPrefix.remove("min_chars"), 0);
+                    int maxChars = XContentMapValues.nodeIntegerValue(indexPrefix.remove("max_chars"), 10);
```

Comment:
this got me wondering whether we should call it min_gram and max_gram like the option on the (edge) ngram tokenizer/filter

Reply:
I've gone back and forth on this a bit.  It would be more consistent, but I think it doesn't make as much sense outside of the context of the ngram tokenizer - if you're reading `{ "index_prefix" : { "min_gram" : 0 } }` it's not obvious what a 'gram' actually is?